[Benjam11n] iP

Benjam11n opened the pull request:
## Joey - Your Personal Task Manager

> "Joey absolutely changed my life. Definitely one of the creations of the 21st century. - Albert Einstein"

A lightweight, intuitive task management application built with JavaFX.

### Overview
Joey is a desktop application designed to help you manage your daily tasks efficiently. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

### Getting Started
1. Download the latest release [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/372)
2. Ensure you have Java 11 or later installed
3. Start managing your tasks!

### For Developers
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public static void main(String[] args) {
    new Joey().run();
}
```


<sub>Disclaimer: Some parts of this PR might have been made up.</sub>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Benjam11n committed:
Rename chatbot from Duke to Joey

Benjam11n committed:
Add echo feature

Benjam11n committed:
Add chat history

Benjam11n committed:
Add History and Task classes

Benjam11n committed:
Refactor chatbot to use History and Task classes

Benjam11n committed:
Add task marking functionality

Benjam11n committed:
Add Todo, Event and Deadline tasks

Benjam11n committed:
Add Text UI Testing

Benjam11n committed:
Add new exception classes

Benjam11n committed:
Add additional error handling

Benjam11n committed:
Add delete task functionality

Benjam11n committed:
Add TaskType and Command enums

Benjam11n committed:
Update EXPECTED.TXT

Benjam11n committed:
Remove unnecessary comments

Benjam11n committed:
Implement saving to a local file

Benjam11n committed:
Merge branch 'branch-Level-7'

Benjam11n committed:
Add Storage class

Benjam11n committed:
Rename History class to TaskList

Benjam11n committed:
Add Ui, Parser and Command classes

Benjam11n committed:
Organise packages

Benjam11n committed:
Merge branch 'branch-level-8'

Benjam11n committed:
Merge branch 'master' into add-gradle-support

Benjam11n committed:
Add dates and times

Benjam11n committed:
Merge branch 'branch-level-8'

Benjam11n committed:
Merge branch 'master' into add-gradle-support

Benjam11n committed:
Setup gradle

Benjam11n committed:
Merge branch 'add-gradle-support'

Benjam11n committed:
Move parsing logic from Command class to Parser class

Benjam11n committed:
Add tests for parser class

Benjam11n committed:
Add tests for Event, Todo and Deadline commands

Benjam11n committed:
Merge branch 'add-junit-tests'

Benjam11n committed:
Add Javadoc comments for Ui class

Benjam11n committed:
Add Javadoc comments for task package

Benjam11n committed:
Add Javadoc comments for storage package

Benjam11n committed:
Add Javadoc comments for parser package

Benjam11n committed:
Add Javadoc comments for exception package

Benjam11n committed:
Fix style errors

Benjam11n committed:
Implement searching for tasks

Benjam11n committed:
Merge branch 'branch-A-JavaDoc'

Benjam11n committed:
Merge branch 'branch-A-CodingStandard'

Benjam11n committed:
Merge branch 'branch-Level-9'

Benjam11n committed:
Update README.md

Benjam11n committed:
Setup checkstyle

Benjam11n committed:
Fix checkstyle errors

Benjam11n committed:
Merge branch 'branch-A-CheckStyle'

Benjam11n committed:
Fix find bug

Benjam11n committed:
Merge branch 'branch-fix-find'

CVincent0907 commented on src/main/java/joey/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+    public static Command parse(String userInput) throws CommandFormatException {
+        String commandWord = userInput.trim().split(" ")[0].toLowerCase();
+
+        switch (commandWord) {
```
The way you arrange the complex parts into a neat one is wonderful. Bravo.

CVincent0907 reviewed (commented):
LSTM!

CVincent0907 commented on src/main/java/joey/command/Command.java:
```diff
@@ -0,0 +1,36 @@
+ * All commands must implement this interface to define their execution behavior.
+ */
+public interface Command {
+    /**
```
Good implementation of Command class to take care of the user input.

heyitssakinah replied:
agreed! great extraction of use cases!

CVincent0907 commented on src/test/java/joey/command/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,48 @@
+    private DeadlineCommand deadlineCommand;
+
+    @BeforeEach
+    void setUp() {
```
Good use of Mock Testing .

CVincent0907 commented on src/main/java/joey/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+    private static final String INVALID_DATE_ERROR_MESSAGE =
+            "Invalid date format. Use YYYY-MM-DD.";
+    private static final String DEFAULT_ERROR_MESSAGE = """
+                        I don't understand this command.
```
Suggestion: Maybe you could take into account case insensitive valid input like toDO ?

CVincent0907 reviewed (commented):
LSTM!

Benjam11n committed:
Implement GUI

Benjam11n committed:
Fix checkstyle errors

Benjam11n committed:
Fix bye command

Benjam11n committed:
Fix find command ui

Benjam11n committed:
Merge branch 'branch-level-10'

heyitssakinah commented on src/main/java/joey/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Represents a command to mark a task in the task list as undone.
+ */
+public class UnmarkCommand implements Command {
```
Minor note on capitalisation: it might be good to standardise if unmark is one or two words? class name suggests one(hence UnmarkCommand), method name below in execute assumes two(hence tasks.unMarkTask). similarly for ui.showUnmarkedtask

heyitssakinah commented on src/main/java/joey/storage/Storage.java:
```diff
@@ -0,0 +1,100 @@
+     * @throws IOException if there is an error reading from the file
+     */
+    @SuppressWarnings("checkstyle:MissingSwitchDefault")
+    public void load(TaskList tasks) throws IOException, IllegalStateException {
```
should the method name be more specific to follow naming conventions?

heyitssakinah commented on src/main/java/joey/ui/DialogBox.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
perhaps the method name could be more specific here? (eg. flipDialogBox) i think it would help explain the method more sufficiently. This issue persists in a few other files as well.

heyitssakinah reviewed (commented):
Overall Looks Good! Well organised code! Just a few parts that can follow naming conventions a bit better.

Benjam11n committed:
Consolidate mark and unmark commands into ToggleCommand

- Combine MarkCommand and UnmarkCommand into a single ToggleCommand class
- Create ToggleType enum to differentiate between mark/unmark operations
- Modify Perser to handle the new command structure

Benjam11n committed:
Add assertions

- Add assertions in the Parser class to check that userInput exists
- Add assertions in the MainWindow class to check that tasks, ui and storage is not null

Benjam11n committed:
Merge pull request #1 from Benjam11n/branch-A-Assertions

Branch a assertions

Benjam11n committed:
Merge branch 'master' into branch-A-CodeQuality

Benjam11n committed:
Fix merge errors

Benjam11n committed:
Merge pull request #2 from Benjam11n/branch-A-CodeQuality

Merge Unmark and Mark commands into Toggle command

Benjam11n committed:
Set up github actions and fix tests

Benjam11n committed:
Merge branch 'branch-A-CI'

Benjam11n committed:
Move gradle.yml to right directory

Benjam11n committed:
Add aliases for commands

Benjam11n committed:
Fix checkstyle errors

Benjam11n committed:
Fix SLAP errors in parser

Benjam11n committed:
Remove prints and text-ui-test

Benjam11n committed:
Add more personality to Joey

Benjam11n committed:
Add styles for error message

Benjam11n committed:
Add more test cases for the Parser class

Benjam11n committed:
Merge branch 'branch-A-BetterGui'

Benjam11n committed:
Merge branch 'branch-A-MoreTesting'

Benjam11n committed:
Fix SLAP issues

Benjam11n committed:
Fix Find command bug

Benjam11n committed:
Update READMD.md and user guide

Benjam11n committed:
Reduce DialogBox padding

Benjam11n committed:
Update user guide image

Benjam11n committed:
Change entry point to Launcher class

Benjam11n committed:
Update dialog-box colors

Benjam11n committed:
Merge branch 'branch-Fix-JavaFX'
